### PR TITLE
Reference preview

### PR DESCRIPTION
**Pull request by stephanebisson**

### CODE REVIEW

**Comment by stephanebisson** [comment]
Notes are similar to references but are mostly unstructured in the article. They won't be supported in this PR.

File: src/components/ReferencePreview.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { h } from 'preact'
+import { useRef } from 'preact/hooks'
+import { useI18n, useSoftkey, useArticleLinksNavigation } from 'hooks'
+
+export const ReferencePreview = ({ reference, lang, close }) => {
+  const i18n = useI18n()
+  const contentRef = useRef()
+  useArticleLinksNavigation('ReferencePreview', lang, contentRef, 1)
+  useSoftkey('ReferencePreview', {
+    right: i18n.i18n('close'),
```
Comment (hueitan) [inline src/components/ReferencePreview.js]: `softkey-close`

File: src/hooks/useArticleLinksNavigation.js [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
   currentPage,
-  onTitleClick,
-  onActionClick
+  linkHandlers = {}
 ) => {
+  const i18n = useI18n()
   const [links, setLinks] = useState([])
   const [currentLink, setCurrentLinkInternal] = useState(null)
 
+  const [showArticlePreview] = usePopup(ArticlePreview, { position: 'bottom' })
```
Comment (hueitan) [inline src/hooks/useArticleLinksNavigation.js]: sometimes the preview only have few lines, then we will have a lot of blank content. do you think 'auto' here better? 

but then the 'loading' effect feel bad =( (or we can do the same as [loading article](https://phabricator.wikimedia.org/T240888), less than 500ms don't show loading)
Screen Shot 2019-12-21 at 12 14 33 AM

File: src/components/ReferencePreview.js (hunk 1 above)
Comment (stephanebisson) [reply src/components/ReferencePreview.js]: Done

File: src/hooks/useArticleLinksNavigation.js (hunk 2 above)
Comment (stephanebisson) [reply src/hooks/useArticleLinksNavigation.js]: Yeah, it turns out 'auto' is the only mode that makes sense. I'll simplify this.

**Comment by stephanebisson** [comment]
I don't know what you mean for `en/Catholic Church` ref 4. I can select all 3 links.
Screen Shot 2019-12-20 at 7 52 47 PM

File: src/hooks/useArticleLinksNavigation.js (hunk 2 above)
Comment (stephanebisson) [reply src/hooks/useArticleLinksNavigation.js]: In our target audience, loading is unlikely to be fast. Whatever we do for the main article loading screen, we should do something similar here.

**Comment by hueitan** [comment]
I just tried on the `ext link`, on browser press `enter/return` key doesn't open the new link, but it does when clicking the `SELECT` key

**Comment by stephanebisson** [comment]
We use `window.open` for external links. Is it possible your browser blocks those? You should try on a device to be sure.

**Comment by hueitan** [comment]
confirm and merge. let's open a separate issue when we found something.

**Pull request merged**